helix-core/src/test/java/org/apache/helix/TestPeriodicRefresh.java

```diff
@@ -0,0 +1,232 @@
+package org.apache.helix;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.helix.api.listeners.MessageListener;
+import org.apache.helix.manager.zk.CallbackHandler;
+import org.apache.helix.manager.zk.ZKHelixManager;
+import org.apache.helix.model.Message;
+import org.apache.helix.zookeeper.api.client.RealmAwareZkClient;
+import org.apache.zookeeper.Watcher;
+import org.testng.Assert;
+import org.testng.annotations.AfterSuite;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static org.apache.helix.HelixConstants.ChangeType.MESSAGE;
+
+
+/**
+ * This class tests that if there are no incoming events, the onMessage method in message listener will be called by message periodic refresh
+ */
+public class TestPeriodicRefresh extends ZkUnitTestBase {
```

Comment:
@kaisun2000 since you raise this concern, please review this test case and verify if this is enough or not : )

Reply:
https://github.com/apache/helix/pull/1000/files#diff-a7737834d48d58f89a95636b50b2858589a27e7043625bffaaa68ece67e3c361R22-R38